build.sbt

```diff
@@ -283,9 +285,36 @@ lazy val scoverageSettings = {
   coverageExcludedPackages := ".+Benchmark.*"
 }
 
+/** Used for publishing `extras`, `api` and `python` jars. Main Spark Job Server assembly is published
+  * as always. */
 lazy val publishSettings = Seq(
+  autoScalaLibrary := false,
+  credentials += Credentials(Path.userHome / ".sbt" / ".credentials"),
+  publishMavenStyle := true,
+  publishTo := Some(sys.env("MVN_PUBLISH_REPO") at sys.env("MVN_PUBLISH_URL")),
   licenses += ("Apache-2.0", url("http://choosealicense.com/licenses/apache/")),
-  bintrayOrganization := Some("spark-jobserver")
+  pomIncludeRepository := { _ => false },
+  /** Since users are encouraged to use dse-spark-dependencies, which provides most of the needed
+    * dependencies, we remove most of the Spark Job Server deps here. Provided, test and blacklisted
+    * deps are removed from resulting poms. */
+  pomPostProcess := { (node: XmlNode) =>
+    new RuleTransformer(new RewriteRule {
+      val pomDependencyBlacklist = Seq("job-server_", "joda-convert", "joda-time")
```

Comment:
if a user ends up running with framework spark-2.0 will this still be ok without joda?

Reply:
we have joda-time in both classpaths so it should be fine. joda-convert is not necessary. It's just for api jars, the main SJS assembly is unchanged